src/backend/sgx/ioctls.rs

```diff
@@ -14,42 +14,41 @@ use sgx::signature::Signature;
 
 const SGX: Group = Group::new(0xA4);
 
-/// IOCTL identifier for ECREATE (see Section 41-21)
+/// ENCLS[CREATE] wrapper
 pub const ENCLAVE_CREATE: Ioctl<Write, &Create<'_>> = unsafe { SGX.write(0x00) };
 
-/// IOCTL identifier for EADD (see Section 41-11)
+/// ENCLS[EADD] wrapper
 pub const ENCLAVE_ADD_PAGES: Ioctl<WriteRead, &AddPages<'_>> = unsafe { SGX.write_read(0x01) };
 
-/// IOCTL identifier for EINIT (see Section 41-35)
+/// ENCLS[EINIT] wrapper
 pub const ENCLAVE_INIT: Ioctl<Write, &Init<'_>> = unsafe { SGX.write(0x02) };
 
-pub const ENCLAVE_SET_ATTRIBUTE: Ioctl<Write, &SetAttribute<'_>> = unsafe { SGX.write(0x03) };
-
-/// SGX_IOC_VEPC_REMOVE_ALL (0x04) is not used by Enarx.
-
-/// SGX_IOC_ENCLAVE_RELAX_PERMISSIONS
-pub const ENCLAVE_RELAX_PERMISSIONS: Ioctl<WriteRead, &RelaxPermissions<'_>> =
+/// ENCLS[EMODPR] wrapper
+pub const ENCLAVE_RESTRICT_PERMISSIONS: Ioctl<WriteRead, &RestrictPermissions<'_>> =
     unsafe { SGX.write_read(0x05) };
 
-/// SGX_IOC_ENCLAVE_RESTRICT_PERMISSIONS
-pub const ENCLAVE_RESTRICT_PERMISSIONS: Ioctl<WriteRead, &RestrictPermissions<'_>> =
+/// SGX_IOC_ENCLAVE_MODIFY_TYPES
+pub const ENCLAVE_MODIFY_TYPES: Ioctl<WriteRead, &ModifyTypes<'_>> =
```

Comment:
I suppose we could also consider making all of these ioctls into associated consts on Enclave, just like we did for DEVICE. I leave it up to your discretion whether or not you think that's an improvement, since you know better than I do about how these end up being used elsewhere.

Reply:
It has crossed my mind. I will make them rather member functions of `Enclave` and pass the enclave around rather than turning into a file instead. But there's enough inter-dependencies on File object that it falls out of scope for this task.